src/anomalib/engine/engine.py

```diff
@@ -225,6 +225,28 @@ def threshold_callback(self) -> _ThresholdCallback | None:
             raise ValueError(msg)
         return callbacks[0] if len(callbacks) > 0 else None
 
+    @property
+    def checkpoint_callback(self) -> ModelCheckpoint | None:
```

Comment:
is this used anywhere else apart from `best_model_path`? If not, would it be an idea to access the checkpoint_callback from the `best_model_path` property?

Reply:
I think it could be useful to have it here, to mirror the functionality of trainer. When using a lightning trainer, the user can directly access the checkpoint callback through the `trainer.checkpoint_callback` property. The current change just extends this behaviour to the engine. We already have similar properties for `thresholding_callback` and `normalization_callback`.

Reply:
Thanks for the clarification. that's fair point